gocourse02/main.go

```diff
@@ -0,0 +1,87 @@
+package main
+
+import (
+	"fmt"
+)
+
+func main() {
+	zookyper := Zookeeper{}.New(false, "Mukola")
+	zookyper.PrintName()
+
+	cage := Cage{}.New(true, 1)
+	fmt.Printf("Cage number %d state =>%t\n", cage.Number, cage.IsBroken)
+	if !zookyper.IsBusy && cage.IsBroken {
+		zookyper.FixCage(cage)
+	}
+	fmt.Printf("Cage number %d state =>%t\n", cage.Number, cage.IsBroken)
+
+	var sirko = Animal{}.New("Sirko", cage, Cat)
+	fmt.Printf("Animal with name %s is sitting in cage number %d type of %s\n", sirko.Name, sirko.Cage.Number, sirko.AnimalType)
+
+	var sirkoSon = Animal{}.New("Sirko junior", cage, Cat)
+	fmt.Printf("Animal with name %s is sitting in cage number %d type of %s\n", sirkoSon.Name, sirkoSon.Cage.Number, sirkoSon.AnimalType)
+
+	var sirkoDaughter = Animal{}.New("", nil, Cat)
+	if sirkoDaughter == nil {
+		fmt.Println("Everything work as expected")
+	}
+
+}
+
+const Cat = "Cat"
+
+// example with struct
+type Zookeeper struct {
+	IsBusy bool
+	Name   string
+}
+
+func (z Zookeeper) New(isBusy bool, name string) *Zookeeper {
```

Comment:
В Go немає механізму конструкторів, як в інших мовах. Але є конвенція, що конструктор - це функція.
```suggestion
func NewZookeeper(isBusy bool, name string) *Zookeeper {
```

Ось почитати коротку відповідь на SO з поясненням https://stackoverflow.com/questions/18125625/constructors-in-go

Або тут на сайті Go https://go.dev/doc/effective_go#composite_literals

Reply:
Попрошу змінити тут і для всіх інших конструкторів.